doc/source/serve/http-servehandle.rst

```diff
@@ -1,92 +1,142 @@
 
-Variable HTTP Routes
-^^^^^^^^^^^^^^^^^^^^
+.. code-block:: python
 
-Ray Serve supports capturing path parameters.  For example, in a call of the form
+    from fastapi import FastAPI
 
-.. code-block:: python
+    app = FastAPI()
 
-    serve.create_endpoint("my_endpoint", backend="my_backend", route="/api/{username}")
+    @serve.deployment(route_prefix="/hello")
+    @serve.ingress(app)
+    class MyFastAPIDeployment:
+        @app.get("/")
+        def root(self):
+            return "Hello, world!"
 
-the ``username`` parameter will be accessible in your backend code as follows:
+    MyFastAPIDeployment.deploy()
+
+Now if you send a request to ``/hello``, this will be routed to the ``root`` method of our Deployment. We can also easily leverage FastAPI to define multiple routes with different HTTP methods:
 
 .. code-block:: python
 
-    def my_backend(request):
-        username = request.path_params["username"]
-        ...
+    from fastapi import FastAPI
+
+    app = FastAPI()
+
+    @serve.deployment(route_prefix="/hello")
+    @serve.ingress(app)
+    class MyFastAPIDeployment:
+        @app.get("/")
+        def root(self):
+            return "Hello, world!"
+
+        @app.post("/{subpath}")
+        def root(self, subpath: str):
+            return f"Hello from {subpath}!"
+
+    MyFastAPIDeployment.deploy()
 
-Ray Serve uses Starlette's Router class under the hood for routing, so type
-conversion for path parameters is also supported, as well as multiple path parameters.  
-For example, suppose this route is used:
+You can also pass in an existing FastAPI app to a deployment to serve it as-is:
 
 .. code-block:: python
-    
-    serve.create_endpoint(
-        "complex", backend="f", route="/api/{user_id:int}/{number:float}")
 
-Then for a query to the route ``/api/123/3.14``, the ``request.path_params`` dictionary 
-available in the backend will be ``{"user_id": 123, "number": 3.14}``, where ``123`` is
-a Python int and ``3.14`` is a Python float.
+    from fastapi import FastAPI
 
-For full details on the supported path parameters, see Starlette's
-`path parameters documentation <https://www.starlette.io/routing/#path-parameters>`_.
+    app = FastAPI()
 
-Custom HTTP response status codes
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    @app.get("/")
+    def f():
+        return "Hello from the root!"
+
+    # ... add more routes, routers, etc. to `app` ...
 
-You can return a `Starlette Response object <https://www.starlette.io/responses/>`_ from your Ray Serve backend code:
+    @serve.deployment(route_prefix="/")
+    @serve.ingress(app)
+    class FastAPIWrapper:
+        pass
+
+This is useful for scaling out an existing FastAPI app with no modifications necessary. You can also combine routes defined this way with routes defined on the Deployment:
```

Comment:
add a note saying existing middlewares, docs, and advanced feature, etc should work as-is